errors.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+class ERR_INVALID_ARG_TYPE extends TypeError {
+  constructor(name, expected, actual) {
+    super(`${name} must be ${expected} got ${actual}`);
+  }
+}
+
+class ERR_NOT_IMPLEMENTED extends Error {
+  constructor(feature) {
+    super(`${feature} not implemented`);
+  }
+}
```

Comment:
by using native `class`, this makes the package not usable in older engines, which precludes its usage by `tape`.

Reply:
Given that the eventual target of this code is the Node.js codebase, it seems like it would be limiting to use JavaScript syntax that targets ancient engines, classes have been around since Node 6 right?

Reply:
Since, in the Node.js codebase, we won't pull in this `errors.js` file (as it already has its own `errors.js`). I don't feel super strongly about this point, I'm more concerned about us avoiding newer syntax like `?.`, `??`, etc. in the actual parser.

Perhaps open a discussion about what our `engines` target should be? and we can come back and modify syntax accordingly.

Reply:
That’s all a fair point. Is there a need to have an error subclass tho, rather than just using an error instance with a code assigned?

Reply:
I'm now setting a `code`, which makes the behavior of `codes. SOME_ERROR` more similar to Node.js -- I'm guessing this is what's implied by Node.js placing the errors on the `codes` key, that the errors created will be populated with `code` equal to their class name.

I would like to continue using a `class`, so that syntax used by a consumer of `errors.js` is the same as in the Node.js codebase:

```js
throw new SOME_ERROR_CODE('info')
```

vs.,

```js
throw SOME_ERROR_CODE('info')
```